packages/rocketchat-language-plugin/server/server.js

```diff
@@ -0,0 +1,82 @@
+const parser = Npm.require('accept-language-parser');
+const languages = Npm.require('languages');
+
+function get_language(languages_codes) {
+
+	const languages_parsed = parser.parse(languages_codes);
+	if (languages_parsed.length===0) {
+		return;
+	}
+	const priority_language = languages_parsed[0].code;
+
+	const priority_language_fullname = languages.getLanguageInfo(priority_language);
+	return priority_language_fullname;
+}
+
+function remove_user_from_automatic_channel() {
+	const user_info = RocketChat.models.Users.findOneById(Meteor.userId());
+	// get the name of channels to which user is added
+	const to_be_removed = user_info.automatic_channels.filter(function(obj) {
+		return obj.blacklisted ==='false';
+	});
+	if (to_be_removed.length!==0) {
```

Comment:
But I think this check is still required because we need ``` to_be_removed``` to remove user from the channel here : 
```
const room = RocketChat.models.Rooms.findOneByIdOrName(to_be_removed[0].name);
RocketChat.removeUserFromRoom(room._id, user);
```
But we can avoid this query: 
``` const user_info = RocketChat.models.Users.findOneById(Meteor.userId()); ```
by simply using:
```
function remove_user_from_automatic_channel(user) {	
const to_be_removed = user.automatic_channels.filter(function(obj) {
```